### PR TITLE
Guard product page when product is null

### PR DESCRIPTION
**Pull request by Codencode**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.1.x
| Description?      | This PR adds a null guard around product-dependent logic in `ProductController` to avoid errors when the product is not available or not properly initialized.<br/>It ensures methods like `initContent()` and breadcrumb generation do not try to access product data when `$this->product` is `null`, making the product page flow more robust in invalid or edge-case contexts.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | In the front office, it tries to access a non-existent article by entering a non-existent product ID.
| UI Tests          | 
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/41292 https://github.com/PrestaShop/PrestaShop/discussions/36868
| Related PRs       | 
| Sponsor company   | Codencode snc

### CODE REVIEW

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -296,102 +297,104 @@ public function initializeCategory(): void
...
+
+            // Presenting pack items
+            $pack_items = $product_for_template['pack'] ? $product_for_template['packItems'] : [];
+            $pack_items = $assembler->assembleProducts($pack_items);
+            $presentedPackItems = [];
+            foreach ($pack_items as $item) {
+                $presentedPackItems[] = $presenter->present(
                     $presentationSettings,
-                    $accessory,
+                    $item,
```
Comment (jf-viguier) [inline controllers/front/ProductController.php:355]: This code come from develop branch not 9.1.x and is not related to your PR, please rebranch from 9.1.x

**Review by jf-viguier — CHANGES_REQUESTED** [review]

File: controllers/front/ProductController.php (hunk 1 above)
Comment (Codencode) [reply controllers/front/ProductController.php:355]: Hello @jf-viguier,
thank you for pointing it out, but I assure you that I created the PR starting from version 9.1.x.
Locally it is correct, I don't understand how this could have happened!
In fact, if I try to rebase, the system tells me that it is up to date!

Comment (Codencode) [reply controllers/front/ProductController.php:355]: I have created a new PR: https://github.com/PrestaShop/PrestaShop/pull/41299
